sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -864,7 +866,8 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
     _VALID_LOSSES = ('least_squares', 'least_absolute_deviation')
 
-    def __init__(self, loss='least_squares', learning_rate=0.1,
+    @_deprecate_positional_args
+    def __init__(self, *, loss='least_squares', learning_rate=0.1,
```

Comment:
I would consider allowing `loss` as positional

Reply:
This kinda brings me back to the n_estimator vs loss issue. Here we have max_iter later in the list.